Automattic/wp-calypso
# State: Introduce action creators and data layer for application passwords

This PR introduces some action creators and a basic data layer for application passwords.

Part of #22912.

To test:
* Checkout this branch
* Verify all tests pass - `npm run test-client application-passwords`
* Load Calypso as a logged in user.
* Call `dispatch( { type: 'APPLICATION_PASSWORDS_REQUEST' } )` in your browser console and verify it performs the expected HTTP request, followed by an `APPLICATION_PASSWORDS_RECEIVE` with the app passwords).
* Call `dispatch( { type: 'APPLICATION_PASSWORD_CREATE', applicationName: 'test here' } )` with an example app name and verify it creates the application password.
* Call `dispatch( { type: 'APPLICATION_PASSWORD_DELETE', appPasswordId: 12345 } )`, where `12345` is the valid ID of an application password you have. Verify it deletes properly.
* Call `dispatch( { type: 'APPLICATION_PASSWORD_CREATE', applicationName: '' } )` and verify you're presented with an error notice.
* Call `dispatch( { type: 'APPLICATION_PASSWORD_DELETE', appPasswordId: 12345 } )`, where `12345` is an invalid ID, verify you're presented with an error notice.
